plugins/woocommerce-admin/client/products/sections/product-shipping-section.tsx

```diff
@@ -60,45 +109,169 @@ export const ProductShippingSection: React.FC = () => {
-								shippingClasses ?? []
-							),
-						] }
-					/>
-					<span className="woocommerce-product-form__secondary-text">
-						{ interpolateComponents( {
-							mixedString: __(
-								'Manage shipping classes and rates in {{link}}global settings{{/link}}.',
-								'woocommerce'
-							),
-							components: {
-								link: (
-									<Link
-										href={ `${ ADMIN_URL }admin.php?page=wc-settings&tab=shipping&section=classes` }
-										target="_blank"
-										type="external"
+			<Card>
+				<CardBody className="product-shipping-section__classes">
+					{ hasResolvedShippingClasses ? (
+						<>
+							<SelectControl
+								label={ __( 'Shipping class', 'woocommerce' ) }
+								{ ...getTextControlProps(
+									getInputProps( 'shipping_class' )
+								) }
+								options={ [
+									...DEFAULT_SHIPPING_CLASS_OPTIONS,
+									...mapShippingClassToSelectOption(
+										shippingClasses ?? []
+									),
+								] }
+							/>
+							<span className="woocommerce-product-form__secondary-text">
+								{ interpolateComponents( {
+									mixedString: __(
+										'Manage shipping classes and rates in {{link}}global settings{{/link}}.',
+										'woocommerce'
+									),
+									components: {
+										link: (
+											<Link
+												href={ `${ ADMIN_URL }admin.php?page=wc-settings&tab=shipping&section=classes` }
+												target="_blank"
+												type="external"
+											>
+												<></>
+											</Link>
+										),
+									},
+								} ) }
+							</span>
+						</>
+					) : (
+						<div className="product-shipping-section__spinner-wrapper">
+							<Spinner />
+						</div>
+					) }
+				</CardBody>
+			</Card>
+
+			<Card>
+				<CardBody className="product-shipping-section__dimensions">
+					{ hasResolvedUnits ? (
+						<>
+							<h4 className="product-shipping-section__subtitle">
+								{ __( 'Dimensions', 'woocommerce' ) }
+							</h4>
+							<p className="woocommerce-product-form__secondary-text">
+								{ __(
+									'Enter the size of the product as you’d put it in a shipping box, including packaging like bubble wrap.',
+									'woocommerce'
+								) }
+							</p>
+							<div className="product-shipping-section__container">
+								<div>
+									<BaseControl
+										id="product_shipping_dimensions_width"
+										className={ inputWidthProps.className }
+										help={ inputWidthProps.help }
 									>
-										<></>
-									</Link>
-								),
-							},
-						} ) }
-					</span>
-				</div>
-			) : (
-				<div className="product-shipping-section__spinner-wrapper">
-					<Spinner />
-				</div>
-			) }
+										<InputControl
+											value={ formatNumber(
+												inputWidthProps.value
+											) }
+											label={ getInterpolatedSizeLabel(
+												__(
+													'Width {{span}}A{{/span}}',
+													'woocommerce'
+												)
+											) }
+											onChange={ ( value: string ) =>
```

Comment:
You also want to pass in the `onBlur` callback to the `InputControl` as otherwise the `help` text doesn't show up.

Reply:
It was covered in the next [PR](https://github.com/woocommerce/woocommerce/pull/34857/files#diff-9200e32ff30bb3150482dcb929ba217f765befd260d42d3ddc24787b2ba5bc0cR202)